massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -1890,4 +1890,84 @@ mod tests {
         // stop the execution controller
         manager.stop();
     }
+
+    #[test]
+    #[serial]
+    fn validate_address() {
+        let vesting = get_initials_vesting(false);
+        // setup the period duration and the maximum gas for asynchronous messages execution
+        let exec_cfg = ExecutionConfig {
+            t0: 100.into(),
+            max_async_gas: 100_000,
+            cursor_delay: 0.into(),
+            initial_vesting_path: vesting.path().to_path_buf(),
+            ..ExecutionConfig::default()
+        };
+        // get a sample final state
+        let (sample_state, _keep_file, _keep_dir) = get_sample_state().unwrap();
+
+        // init the storage
+        let mut storage = Storage::create_root();
+        // start the execution worker
+        let (mut manager, controller) = start_execution_worker(
+            exec_cfg.clone(),
+            sample_state.clone(),
+            sample_state.read().pos_state.selector.clone(),
+        );
+        // initialize the execution system with genesis blocks
+        init_execution_worker(&exec_cfg, &storage, controller.clone());
+
+        // keypair associated to thread 0
+        let keypair =
+            KeyPair::from_str("S1JJeHiZv1C1zZN5GLFcbz6EXYiccmUPLkYuDFA3kayjxP39kFQ").unwrap();
+        // let address = Address::from_public_key(&keypair.get_public_key());
+
+        // load bytecode
+        // you can check the source code of the following wasm file in massa-unit-tests-src
+        let bytecode = include_bytes!("./wasm/validate_address.wasm");
+
+        // create the block containing the erroneous smart contract execution operation
+        let operation =
+            create_execute_sc_operation(&keypair, bytecode, BTreeMap::default()).unwrap();
+        storage.store_operations(vec![operation.clone()]);
+        let block = create_block(KeyPair::generate(), vec![operation], Slot::new(1, 0)).unwrap();
+
+        // store the block in storage
+        storage.store_block(block.clone());
+
+        // set our block as a final block
+        let mut finalized_blocks: HashMap<Slot, BlockId> = Default::default();
+        finalized_blocks.insert(block.content.header.content.slot, block.id);
+        let block_store = vec![(block.id, storage.clone())].into_iter().collect();
+        controller.update_blockclique_status(finalized_blocks, Default::default(), block_store);
+        std::thread::sleep(
+            exec_cfg
+                .t0
+                .saturating_add(MassaTime::from_millis(50))
+                .into(),
+        );
+
+        let events = controller.get_filtered_sc_output_event(EventFilter::default());
+        // match the events
+        assert!(!events.is_empty(), "2 events were expected");
```

Comment:
Why not matching directly the wanted length?

Reply:
Fixed